apps/camera_pipe/camera_pipe_generator.cpp

```diff
@@ -500,11 +500,15 @@ void CameraPipe::generate() {
         Expr out_width = processed.width();
         Expr out_height = processed.height();
 
-        // In HVX 128, we need 4 threads to saturate HVX with work,
-        // on other devices, we might need many threads.
+        // Depending on the HVX generation, we need 2 or 4 threads
+        // to saturate HVX with work.
         Expr strip_size;
         if (get_target().has_feature(Target::HVX)) {
-            strip_size = processed.dim(1).extent() / 4;
+            if (get_target().features_any_of({Target::HVX_v65, Target::HVX_v66})) {
```

Comment:
Should this condition be negated, so its future proof?

Also, doesn't this vary between generations? I think maybe we should just unconditionally use 4. It's not a big deal to oversubscribe the contexts, maybe overhead is negligibly higher? But at the same time, we often split more than necessary anyways to get the working sets small enough to fit in the cache.

Reply:
You are right. To keep things simple, I'll switch the code to use 4 threads always.